testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js

```diff
@@ -2556,6 +2556,47 @@ QUnit.testInActiveWindow("input should be focused after click on field (searchEn
     assert.ok($input.is(":focus"), "input was focused");
 });
 
+QUnit.test("select all is not applied when filterable items are selected", function(assert) {
+    var items = ["111", "222", "333"],
+        $element = $("#tagBox").dxTagBox({
+            searchTimeout: 0,
+            items: items,
+            searchEnabled: true,
+            opened: true,
+            showSelectionControls: true,
+            selectAllMode: "allPages"
+        }),
+        instance = $element.dxTagBox("instance"),
+        $input = $element.find("input");
+
+    keyboardMock($input).type("1");
+
+    $(".dx-tagbox-popup-wrapper .dx-list-select-checkbox.dx-checkbox").trigger("dxclick");
+
+    assert.equal(instance.option("selectedItems").length, 1, "selected items count");
+});
+
+QUnit.test("filter should not be cleared when no focusout and no item selection happened", function(assert) {
+    var items = ["111", "222", "333"],
+        $element = $("#tagBox").dxTagBox({
+            searchTimeout: 0,
+            items: items,
+            searchEnabled: true,
+            opened: true,
+            showSelectionControls: true,
+            selectAllMode: "allPages"
+        }),
+        $input = $element.find("input");
+
+    var keyboard = keyboardMock($input);
+    keyboard.type("1");
+    keyboard.press("esc");
+
+    $input.trigger("dxclick");
+
+    assert.equal($(".dx-item").length, 1, "items count of list");
```

Comment:
I think would be better replace all occurrences of `dx-item` with constant

Reply:
What a terrible idea - using constants in tests....
Tests should be easy to read

Reply:
So how using terrible selectors help make tests easy to read?
Do not confuse poorly written test (that hard to read) with decreasing a chance to make a typo with a class name.

Reply:
If you make a typo in the class name, your test should fail, shouldn't it?

Reply:
Yep. Then you try to make it green, fix the bug. But test still fails and you'll lose the time to locate the problem.